proto/cacheclient.proto

```diff
@@ -403,6 +404,8 @@ message _ListRemoveResponse {}
 
 message _ListFetchRequest {
   bytes list_name = 1;
+  optional uint32 count = 2;
+  optional sint32 start_index = 3;
```

Comment:
Summary from the conversation,
1. We can add a `end_index` in the future, then enforce either count or end_index. This would _probably_ be enforced based on the sdk language as some use count and others use end_index
2. `Step`, a signed int, can be added later if customers ask for the ability to reverse through the list and/or change the stride to get every nth element

Reply:
i'm not sure if we need to explicitly mark these as optional in the proto; I think for other "optional" fields we have been just relying on recognizing when they have the protobuf "default" value.  Can you check me on this, @kvcache ?

Reply:
According to the 1st line in this file, this is `proto3` syntax.

Absent field value semantics are defined in the `proto3` language guide https://protobuf.dev/programming-guides/proto3/#default 

Some observations:
* `optional` does not mean anything material in proto3.
* Some code generators do seem to use it.
* Some protoc versions do not accept `optional` in proto3 (some linting thing) and it has blocked our pipelines before (internal pipelines iirc) when people marked fields `optional`.
* None of our ~60 request/response message definitions in this file use `optional`.

Do you have a concrete reason to use `optional` in these fields?

Reply:
https://github.com/protocolbuffers/protobuf/releases/tag/v3.15.0
Optional is on by default, so not necessary, removing. Looks like I was looking at proto2 docs as it's the second link for `proto optional`